### ISSUE TITLE
몬테카를로 드랍아웃

### ISSUE THREAD

**Issue by vmtmxmf5** [vmtmxmf5/Python-ML-DNN#75]
## MonteCarlo Dropout
: test 할 때도 dropout을 켜서, 다양한 예측값을 만들어낸다.   
그 예측값들로 분포를 만들어서, 클래스 분류 결과의 불확실성을 추정한다.
(몬테카를로 : 랜덤 샘플링해서 numerical quantity(예측값 등)의 distribution을 얻는 기법) 

- range(하이퍼파라미터) 예측값으로 분포를 만들기 위함
- 2^N 개의 네트워크 경우의 수 중 일부 예측값의 평균 모델이 특정 클래스를 답으로 확신하는 정도
- 2^N 개의 네트워크 경우의 수 중 일부의 분산 분산이 클 수록 예측이 불확실하다는 의미

```sage
class MCDropout(tf.keras.layers.Dropout):
    def call(self, inputs):
        return super().call(inputs, training=True)
    
y_probas = np.stack([model(X_test) for sample in range(100)]) #하이퍼파라미터 range

y_proba = y_probas.mean(axis=0) # 평균
y_std = y_probas.std(axis=0) # 분산
```
